jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/Asn1EncodedDataRouter.java

```diff
@@ -41,409 +38,412 @@
-
-   private static final String BYTES = "bytes";
-
-   private static final String MESSAGE_FRAME = "MessageFrame";
-
-   private static final String ERROR_ON_SDX_DEPOSIT = "Error on SDX deposit.";
-
-   public static class Asn1EncodedDataRouterException extends Exception {
-
-      private static final long serialVersionUID = 1L;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
 
-      public Asn1EncodedDataRouterException(String string) {
-         super(string);
-      }
+@Slf4j
+public class Asn1EncodedDataRouter extends AbstractSubscriberProcessor<String, String> {
 
-   }
+    private static final String BYTES = "bytes";
+    private static final String MESSAGE_FRAME = "MessageFrame";
+    private static final String ERROR_ON_SDX_DEPOSIT = "Error on SDX deposit.";
+
+    public static class Asn1EncodedDataRouterException extends Exception {
+        private static final long serialVersionUID = 1L;
+
+        public Asn1EncodedDataRouterException(String string) {
+            super(string);
+        }
+    }
+
+    private final OdeProperties odeProperties;
+    private final Asn1CoderTopics asn1CoderTopics;
+    private final JsonTopics jsonTopics;
+
+    private final MessageProducer<String, String> stringMsgProducer;
+    private final OdeTimJsonTopology odeTimJsonTopology;
+    private final Asn1CommandManager asn1CommandManager;
+    private final boolean dataSigningEnabledRSU;
+    private final boolean dataSigningEnabledSDW;
+
+    public Asn1EncodedDataRouter(OdeProperties odeProperties,
+                                 OdeKafkaProperties odeKafkaProperties,
+                                 Asn1CoderTopics asn1CoderTopics,
+                                 JsonTopics jsonTopics,
+                                 SDXDepositorTopics sdxDepositorTopics) {
+        super();
+
+        this.odeProperties = odeProperties;
+        this.asn1CoderTopics = asn1CoderTopics;
+        this.jsonTopics = jsonTopics;
+
+        this.stringMsgProducer = MessageProducer.defaultStringMessageProducer(odeKafkaProperties.getBrokers(),
+                odeKafkaProperties.getProducerType(),
+                odeKafkaProperties.getDisabledTopics());
+
+        this.asn1CommandManager = new Asn1CommandManager(odeProperties, odeKafkaProperties, sdxDepositorTopics);
+
+        this.dataSigningEnabledRSU = System.getenv("DATA_SIGNING_ENABLED_RSU") != null && !System.getenv("DATA_SIGNING_ENABLED_RSU").isEmpty()
+                ? Boolean.parseBoolean(System.getenv("DATA_SIGNING_ENABLED_RSU"))
+                : false;
+        this.dataSigningEnabledSDW = System.getenv("DATA_SIGNING_ENABLED_SDW") != null && !System.getenv("DATA_SIGNING_ENABLED_SDW").isEmpty()
+                ? Boolean.parseBoolean(System.getenv("DATA_SIGNING_ENABLED_SDW"))
+                : true;
+
+        odeTimJsonTopology = new OdeTimJsonTopology(odeKafkaProperties);
+        if (!odeTimJsonTopology.isRunning()) {
```

Comment:
It won't. I thought I'd already added a note about this change, but I guess it got lost.

The `OdeTimJsonTopology` is only ever created on line 94 (or what was 91 before). A new  `OdeTimJsonTopology` is only instantiated when a new `Asn1EncodedDataRouter` is created. There's only one place/time we create a new `Asn1EncodedDataRouter` ([here](https://github.com/CDOT-CV/jpo-ode/blob/a80dcb958068ade0eb050a1c84af3cdb173524c1/jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/AsnCodecRouterServiceController.java#L59)). With only one way to create a new  `OdeTimJsonTopology` object, we do not need for making the property in `Asn1EncodedDataRouter` static, so there's no need to check if an  `OdeTimJsonTopology` is null. It will always be created on line 94, so there's no chance of null.

In the future, if we find that we're creating a lot of `OdeTimJsonTopology` instances that are expensive, we can evaluate our options then. For example, we could use a [Spring Singleton Bean](https://www.baeldung.com/spring-boot-singleton-vs-beans) if we only want one. These don't look like they're concerns yet, so it's not worth our time to implement that (or other patterns) or maintain the current overly complex code. You Ain't Gonna Need It ™️